### PR TITLE
feat: authentication for the new client

### PR DESCRIPTION
**Pull request by amritk**
In this PR we create schemas and store data for the following auth types
- http basic
- http bearer
- apiKey cookie
- apiKey header
- apiKey query

What this PR does not do:
- oauth2
- openIdConnect
- two way binding with references
- inject the correct data on request

These will come tomorrow!

### CODE REVIEW

File: packages/client-app/src/views/Request/RequestSection/RequestAuth.vue [hunk 1]
```diff
@@ -5,157 +5,202 @@ import DataTableHeader from '@/components/DataTable/DataTableHeader.vue'
 import DataTableInput from '@/components/DataTable/DataTableInput.vue'
 import DataTableRow from '@/components/DataTable/DataTableRow.vue'
 import ViewLayoutCollapse from '@/components/ViewLayout/ViewLayoutCollapse.vue'
+import { useWorkspace } from '@/store/workspace'
 import { ScalarButton, ScalarIcon, ScalarListbox } from '@scalar/components'
-import { computed, ref } from 'vue'
+import { capitalize, computed, ref } from 'vue'
```
Comment (antlio) [inline packages/client-app/src/views/Request/RequestSection/RequestAuth.vue:10]: til!

File: packages/client-app/src/views/Request/RequestSection/RequestAuth.vue [hunk 2]
```diff
@@ -5,157 +5,202 @@ import DataTableHeader from '@/components/DataTable/DataTableHeader.vue'
...
+  securitySchemes,
+  securitySchemeMutators,
+} = useWorkspace()
+
+const columnLayout = computed(() =>
+  activeScheme.value?.type === 'oauth2' ||
+  activeScheme.value?.type === 'openIdConnect'
+    ? ['', 'auto']
+    : [''],
+)
```
Comment (antlio) [inline packages/client-app/src/views/Request/RequestSection/RequestAuth.vue:29]: nice 👍

File: packages/components/src/components/ScalarListbox/ScalarListbox.vue [hunk 3]
```diff
@@ -59,7 +59,7 @@ const variants = cva({
       </ListboxButton>
       <template #floating="{ width }">
         <ListboxOptions
-          class="relative flex w-40 flex-col p-0.75"
+          class="relative flex w-40 flex-col rounded border p-0.75"
```
Comment (antlio) [inline packages/components/src/components/ScalarListbox/ScalarListbox.vue:62]: good catch!

**Review by antlio — APPROVED** [review]
incredible! 🛸

**Review by marclave — APPROVED** [review]
fr fr this is nutty work @amritk

**Pull request merged**